<img src="images/logodwengo.png" alt="Banner" style="width: 150px;"/>

<div style='color: #690027;' markdown="1">
    <h1>PYTHAGORAS</h1> 
</div>

<div class="alert alert-block alert-success">
De stelling van Pythagoras leerde je reeds in de wiskundeles. Met deze notebook zal je het rekenwerk van de oefeningen automatiseren. Bij het maken van de oefeningen in de wiskundeles kan je je dan volop concentreren op het denkwerk.
</div>

<div class="alert alert-block alert-info">
Er zijn eigenlijk maar twee types van oefeningen bij de stelling van Pythagoras:<br>
    - de lengte van de schuine zijde berekenen als de rechthoekszijden gekend zijn;<br>
    - de lengte van een rechthoekszijde berekenen als de schuine zijde en de andere rechthoekszijde gekend zijn.<br>
    Had je dit al door? Dan deed je aan <b>patroonherkenning</b>, een concept van computationeel denken.
</div>

Dat betekent dat je voor de automatisatie slechts 2 functies moet definiëren. Een voor elke type oefening.

### Nodige modules importeren

In [1]:
import math         # om vierkantswortels en pi te kunnen gebruiken

<div class="alert alert-block alert-warning"> 
Meer voorbeelden van het gebruik van de module math vind je in de notebook 'Rekenen'.<br>
    In plaats van de module math kan je ook de module NumPy gebruiken. Numpy komt ook verder in deze notebook aan bod.
</div>

<div style='color: #690027;' markdown="1">
    <h2>1. Functies</h2> 
</div>

<div style='color: #690027;' markdown="1">
    <h3>1.1 De lengte van de schuine zijde berekenen als de rechthoekszijden gekend zijn</h3> 
</div>

### Opdracht

Schrijf een script waarin je een functie `pythagoras` definieert. <br>
Het script vraagt de gebruiker om de lengtes van de twee rechthoekszijden van een rechthoekige driehoek in te geven. Via de functie `pythagoras` zal het script de lengte van de schuine zijde berekenen. Het programma eindigt met het tonen van de lengte van de schuine zijde aan de gebruiker.<br>
<br>
Test het script uit en verbeter indien nodig.

In [2]:
# voorbeeldscript
# functie definiëren
def pythagoras(a, b):
    """Schuine zijde rechthoekige driehoek met rechthoekszijden a en b."""
    schuinezijde = math.sqrt(a**2 + b**2)
    return schuinezijde

# input
print("Geef de lengtes in van de twee rechthoekszijden van een rechthoekige driehoek.")    
rhz1 = float(input("Eerste rechthoekszijde: "))
rhz2 = float(input("Tweede rechthoekszijde: "))

# input verwerken
c = pythagoras(rhz1, rhz2)    # functie wordt opgeroepen en geeft schuine zijde terug,  variabele c refereert aan teruggegeven waarde

# output
print("\n De schuine zijde is", c)

Geef de lengtes in van de twee rechthoekszijden van een rechthoekige driehoek.
Eerste rechthoekszijde: 3
Tweede rechthoekszijde: 7

 De schuine zijde is 7.615773105863909


<div class="alert alert-block alert-info">
Het bovenstaande script kunnen we opsplitsen in 4 onderdelen:
    <ol>
        <li>definitie functie</li>
        <li>input</li>
        <li>verwerking</li>
        <li>output</li>
    </ol>
Voor het definiëren van een functie heb je enkel een definitie nodig. Eens je de functie wilt gaan testen en/of gebruiken zal je ook invoer nodig hebben en zal er hopelijke een uitvoer verschijnen.
    <img src="images/invoerverwerkinguitvoerdwengo.png" alt="Banner" style="width: 150px;"/>
     <img src="images/decompositiedwengo.png" alt="Banner" style="width: 150px;"/>
    <img src="images/abstractiedwengo.png" alt="Banner" style="width: 150px;"/>
</div>

<div class="alert alert-block alert-warning"> 
Uitleg over input en output vind je terug in de notebook "InputEnOutput".
</div>

<div style='color: #690027;' markdown="1">
    <h3>1.2 De lengte van een rechthoekszijde berekenen als de schuine zijde en de andere rechthoekszijde gekend zijn</h3> 
</div>

### Opdracht

Schrijf een script waarin je een functie `pythagoras2` definieert. <br>
Het script vraagt de gebruiker om de lengtes van de schuine zijde en een rechthoekszijde van een rechthoekige driehoek in te geven. Via de functie `pythagoras` zal het script de lengte van de andere rechthoekszijde berekenen. Het programma eindigt met het tonen van de lengte van de andere rechthoekszijde aan de gebruiker.<br>
<br>
Test het script uit en verbeter indien nodig.

In [3]:
# voorbeeldscript
# functie definiëren
def pythagoras2(a, c):
    """Rechthoekszijde rechthoekige driehoek met schuine zijde c en rechthoekszijde a."""
    rechthoekszijde = math.sqrt(c**2 - a**2)
    return rechthoekszijde

# input
print("Geef de lengtes in van de rechthoekszijde en de schuine zijde van een rechthoekige driehoek.")    
rhz1 = float(input("Rechthoekszijde: "))
sz = float(input("Schuine zijde: "))

# input verwerken
rhz2 = pythagoras2(rhz1, sz)    # functie wordt opgeroepen en geeft rechthoekszijde terug, variabele rhz2 refereert aan teruggegeven waarde

# output
print("\n De andere rechthoekszijde is", rhz2)

Geef de lengtes in van de rechthoekszijde en de schuine zijde van een rechthoekige driehoek.
Rechthoekszijde: 4
Schuine zijde: 5

 De andere rechthoekszijde is 3.0


<div style='color: #690027;' markdown="1">
    <h2>2. Voorbeeld</h2> 
</div>

### Voorbeeld 2.1

Bereken de oppervlakte van een rechthoekige driehoek met een rechthoekszijde van 8 cm en een schuine zijde van 14 cm.

In [4]:
import math

# oppervlakte kan berekend worden met functie uit vorige notebook
def oppdriehoek(b,h):
    """Oppervlakte van driehoek met basis b en hoogte h."""
    opp = b * h /2
    return opp

# oppervlakte driehoek kan je berekenen als de basis en de hoogte gekend zijn, dus hier de twee rechthoekszijden
# andere rechthoekszijde berekenen met Pythagoras2
def pythagoras2(a, c):
    """Rechthoekszijde rechthoekige driehoek met schuine zijde c en rechthoekszijde a."""
    rechthoekszijde = math.sqrt(c**2 - a**2)
    return rechthoekszijde

rhz1 = 8                           # gegeven (invoer)
sz = 14                            # gegeven (invoer)
rhz2 = pythagoras2(rhz1, sz)       # andere rechthoekszijde (verwerking)
opp = oppdriehoek(rhz1, rhz2)      # gezochte oppervlakte (verwerking)

print("De oppervlakte van de driehoek is", opp, "cm^2")       # uitvoer

De oppervlakte van de driehoek is 45.95650117230423 cm^2


<div style='color: #690027;' markdown="1">
    <h2>3. Oefeningen</h2> 
</div>

### Oefening 2.1

Het stratenplan van New York bestaat vooral uit straten die loodrecht op elkaar staan.<br>
Bekijk de figuren en bereken m.b.v. Python-code hoe ver het is in vogelvlucht van de Empire State Building naar Times Squares.
<img src="images/kaartnewyork.png" alt="Banner" style="width: 400px;"/> <br>
<img src="images/gpsnewyork.png" alt="Banner" style="width: 400px;"/>

In [5]:
# voorbeeldscript
# rechthoekszijden zijn 600 meter en 1000 m
pythagoras(600, 1000)

1166.19037896906

Antwoord:

Antwoord: Het is ongeveer 1166 meter in vogelvlucht.

### Oefening 2.2

Als je een ladder veilig tegen de muur wilt plaatsen, dan gebruik je best de '4 t.o.v. 1'- regel. Dat betekent dat voor elke 4 eenheden hoogte, de ladder 1 eenheid van de muur moet staan.  <br><br>
Stel dat een ladder veilig tegen de muur staat volgens de '4 t.o.v. 1'-regel en de muur raakt op 5 meter boven de grond.
Hoe lang is de ladder? Bereken met Python-code.

In [6]:
# voorbeeldscript
rhz1 = 5
rhz2 = 5 / 4
pythagoras(rhz1, rhz2)

5.153882032022076

Antwoord:

Antwoord: De ladder is 5,15 meter lang.

### Oefening 2.2

Een ladder van 10 meter lang staat tegen een muur op 1,5 meter van de muur. Staat de ladder veilig?
Voer je rekenwerk uit met Python-code.

In [7]:
# voorbeeldscript
sz = 10
rhz1 = 1.5
rhz2 = pythagoras2(rhz1, sz)
print(rhz2, rhz1 * 4)  

9.886859966642595 6.0


Antwoord:

Antwoord: Neen, de ladder staat niet veilig. Hij reikt te hoog en staat te steil.

<img src="images/cclic.png" alt="Banner" align="left" style="width:100px;"/><br><br>
Notebook Python in wiskunde, zie Computationeel denken - Programmeren in Python van <a href="http://www.aiopschool.be">AI Op School</a>, van F. wyffels, B. Van de Velde & N. Gesquière is in licentie gegeven volgens een <a href="http://creativecommons.org/licenses/by-nc-sa/4.0/">Creative Commons Naamsvermelding-NietCommercieel-GelijkDelen 4.0 Internationaal-licentie</a>. 